### ISSUE TITLE
Change snippets directory specification

### ISSUE THREAD

**Issue by einarnn** [CiscoDevNet/ncc#4]
From aradford123:

> I interpreted snippets a bit differently.
>
> It was to specify a snippet directory, not the path to /snippets.
```
named_filters = Environment(loader=FileSystemLoader(
    '%s/%s/filters' % (NCC_DIR,args.snippets)))
named_templates = Environment(loader=FileSystemLoader(
    '%s/%s/editconfigs' % (NCC_DIR, args.snippets)))
```

**Comment by einarnn** [comment]
I would suggest just doing:

```
if args.snippets:
    snippets = args.snippets
else:
    snippets = NCC_DIR + '/snippets'

named_filters = Environment(loader=FileSystemLoader(
    '%s/filters' % snippets))
named_templates = Environment(loader=FileSystemLoader(
    '%s/editconfigs' % snippets))
```

This gives the flexibility to locate the snippets directory anywhere and would not mandate it living in NCC_DIR.

**Comment by aradford123** [comment]
ok.  that works.

**Comment by aradford123** [comment]
can we provide a way to "default" the snippets.

I was thinking something like having:

ncc/
├── snippets-xe
├── snippets-xr
but also allowing a symbolic link from snippets to one of the above.

How do we handle open models then?

snippets-open/config-snippet?  That would map to snippets-open/snippets/config-snippet

I would use open and xe models at the same time.  At present i have mixed them together but should really break them out?

**Comment by einarnn** [comment]
What do you do when you have a snippet that uses an open model and a native model in the same request?

**Comment by aradford123** [comment]
that would be where you could specify the "prefix" ("xe", "xr" "open").  Doubt you would use "xr" and "xe" in same request.

open/interfaces xe/oper-data etc

**Comment by einarnn** [comment]
Addressed in #11 .
